File: learning-service/src/main/java/com/learning/domain/model/LessonSession.java
package com.learning.domain.model;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class LessonSession {
    private String id;
    private String studentId;
    private String lessonId; // Reference to the lesson concept
    private int currentChallengeIndex;
    private int correctAnswers;
    private int wrongAnswers;
    private int heartsRemaining;
    private LessonStatus status;
}

File: learning-service-php/src/Domain/Model/LessonSession.php
<?php

namespace App\Domain\Model;

class LessonSession {
    public function __construct(
        public string $id,
        public string $studentId,
        public string $lessonId,
        public string $status = 'IN_PROGRESS',
        public int $currentChallengeIndex = 0,
        public int $heartsRemaining = 5,
        public int $xpGained = 0
    ) {}

    public function isFinished(): bool {
        return $this->status !== 'IN_PROGRESS';
    }
}
